#! /bin/sh /usr/share/dpatch/dpatch-run
## 11_debian_version.dpatch by  <jblache@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Embed Debian package revision and MODE into the binary.

@DPATCH@
diff -urNad arcload-0.5~/Makefile arcload-0.5/Makefile
--- arcload-0.5~/Makefile	2007-09-11 16:13:26.000000000 +0000
+++ arcload-0.5/Makefile	2007-09-11 16:14:09.000000000 +0000
@@ -3,6 +3,9 @@
 # Copyright 2004 Stanislaw Skowronek
 #  
 
+# Debian version
+DEBVER = $(shell dpkg-parsechangelog | grep ^Version | cut -d' ' -f 2)
+
 # Default MODE
 MODE ?= M64
 
@@ -24,5 +27,5 @@
 		MODE=${MODE} $(MAKE) -C $$i $@; \
 	done
 	for i in $(BUILD_DIRS); do \
-		CC=${CC} LD=${LD} MODE=${MODE} $(MAKE) -C $$i $@; \
+		CC=${CC} LD=${LD} MODE=${MODE} DEBVER=${DEBVER} $(MAKE) -C $$i $@; \
 	done
diff -urNad arcload-0.5~/loader/Makefile arcload-0.5/loader/Makefile
--- arcload-0.5~/loader/Makefile	2006-01-20 04:10:04.000000000 +0000
+++ arcload-0.5/loader/Makefile	2007-09-11 16:15:00.000000000 +0000
@@ -2,12 +2,12 @@
 # Copyright 2004 Stanislaw Skowronek
 #
 ifeq ($(MODE),M32)
-CFLAGS = -O -march=mips3 -mabi=32 -Wall -mno-abicalls -G 0 -fno-pic -fno-builtin -I.. -D$(MODE)
+CFLAGS = -O -march=mips3 -mabi=32 -Wall -mno-abicalls -G 0 -fno-pic -fno-builtin -I.. -D$(MODE) -DDEBVER="\"$(DEBVER) M32\""
 ASFLAGS= -march=mips3 -mabi=32 -mno-abicalls -G 0 -fno-pic -D$(MODE)
 OUTPUTFMT = elf32-tradbigmips
 TARGET = ../arcload.ecoff
 else
-CFLAGS = -O -march=mips3 -mabi=64 -Wall -mno-abicalls -G 0 -fno-pic -fno-builtin -I.. -D$(MODE)
+CFLAGS = -O -march=mips3 -mabi=64 -Wall -mno-abicalls -G 0 -fno-pic -fno-builtin -I.. -D$(MODE) -DDEBVER="\"$(DEBVER) M64\""
 ASFLAGS= -march=mips3 -mabi=64 -mno-abicalls -G 0 -fno-pic -D$(MODE)
 OUTPUTFMT = elf64-tradbigmips
 TARGET = ../arcload
diff -urNad arcload-0.5~/loader/main.c arcload-0.5/loader/main.c
--- arcload-0.5~/loader/main.c	2006-01-20 07:11:20.000000000 +0000
+++ arcload-0.5/loader/main.c	2007-09-11 16:15:43.000000000 +0000
@@ -270,7 +270,7 @@
 
 	unsigned long arena_start;
 
-	printf("ARCLoad version " VERSION " (c) 2004-5 Stanislaw Skowronek\n\r");
+	printf("ARCLoad version " VERSION " (Debian " DEBVER ") (c) 2004-5 Stanislaw Skowronek\n\r");
 
 #ifdef M64
 	fixup_trampolines();
